src/cfn-resource-mappings.ts

```diff
@@ -116,7 +116,19 @@ export function mapToCfnResource(
             const mName = moduleName(typeName).toLowerCase();
             const pType = pulumiTypeName(typeName);
             const awsModule = aws as any;
-            return new awsModule[mName][pType](logicalId, props, options);
+
+            // Workaround until TODO[pulumi/pulumi-aws-native#1816] is resolved.
```

Comment:
Without this none of the CDK CustomResources I tried deploy successfully. Their all nested too deep for the 64 char limits of Lambda Functions and IAM roles